src/UI/UIView.cpp

```diff
@@ -45,596 +54,905 @@ void UIView::draw_ui_callback()
+      std::string movie_button_label = m_controller_app.get_movie_state_label();
+      if (movie_button_label == "Write Movie")
       {
-        m_controller_app.run_movie();
+        if (ImGui::Button(movie_button_label.c_str()))
+        {
+          m_controller_app.get_simulation().set_animation_state();
+          m_controller_app.run_movie(true);
+        }
+      }
+      else
+      {
+        if (ImGui::Button(movie_button_label.c_str()))
+        {
+          m_controller_app.stop_movie();
+        }
       }
 #endif
-    }
-    ImGui::EndTabItem();
-  }
 
-  if (ImGui::BeginTabItem("Clipping"))
-  {
-    ImGui::Text("Axis aligned clip");
-    glm::vec2 bound = { m_clip_min.x * 100 ,m_clip_max.x * 100 };
-    ImGui::DragFloatRange2("X", &bound.x, &bound.y, 0.1f, 0.0f, 100.0f, "Min: %.1f %%", "Max: %.1f %%");
-    m_clip_min.x = bound.x / 100;
-    m_clip_max.x = bound.y / 100;
-
-    bound = { m_clip_min.y * 100 ,m_clip_max.y * 100 };
-    ImGui::DragFloatRange2("Y", &bound.x, &bound.y, 0.1f, 0.0f, 100.0f, "Min: %.1f %%", "Max: %.1f %%");
-    m_clip_min.y = bound.x / 100;
-    m_clip_max.y = bound.y / 100;
-
-    bound = { m_clip_min.z * 100 ,m_clip_max.z * 100 };
-    ImGui::DragFloatRange2("Z", &bound.x, &bound.y, 0.1f, 0.0f, 100.0f, "Min: %.1f %%", "Max: %.1f %%");
-    m_clip_min.z = bound.x / 100;
-    m_clip_max.z = bound.y / 100;
-
-    if (ImGui::Button("Reset")) {
-      m_clip_min = glm::vec3(0.0f);
-      m_clip_max = glm::vec3(1.0f);
+      ImGui::EndTabItem();
     }
 
-    ImGui::Checkbox("Custom Clipping plane", &m_useCustomClipPlane);
-    if (m_useCustomClipPlane) {
-      ImGui::SliderAngle("Pitch", &m_clip_ypr.y, -90, 90);
-      ImGui::SliderAngle("Roll", &m_clip_ypr.z, -180, 180);
-
-      ImGui::SliderFloat("Position X", &m_clip_pos.x, -0.5, 0.5);
-      ImGui::SliderFloat("Position y", &m_clip_pos.y, -0.5, 0.5);
-      ImGui::SliderFloat("Position z", &m_clip_pos.z, -0.5, 0.5);
-      if (ImGui::Button("Reset##Reset2")) {
-        m_clip_ypr = glm::vec3(0.0f);
-        m_clip_pos = glm::vec3(0.0f);
+
+    ImGui::EndTabBar();
+
+    if (m_time_frame_edited)
+    {
+      ImGui::OpenPopup("Time frame Changed##successchange");
+      ImGui::SetNextWindowSize(ImVec2(350, 400), ImGuiCond_FirstUseEver);
+      if (ImGui::BeginPopupModal("Time frame Changed##successchange", &m_time_frame_edited))
+      {
+        ImGui::Text("Changes saved");
+        if (ImGui::Button("Ok"))
+        {
+          m_time_frame_edited = false;
+          ImGui::CloseCurrentPopup();
+        }
+        ImGui::EndPopup();
       }
 
     }
-    ImGui::EndTabItem();
-  }
-  ImGui::EndTabBar();
 
-  //file loading
-  if (m_file_dialog_open)
-  {
-    ImGui::OpenPopup("Open File");
-    m_file_dialog_open = false;
-  }
+    if (m_camera_name_window_open)
+    {
+      std::string modal_window_name;
 
+      switch (m_camera_button_action)
+      {
+      case ADD:
+        modal_window_name = "Add Simulation State";
```

Comment:
what is a simulation state? I'm seeing animation, simulation, camera, time frame, and poi all used and some of them seem to be referring to the same/similar concepts - could this vocabulary use be clarified/tightened up?

Reply:
Simulation state is a key frame in the animation where the camera and slices have a particular state. The animation is the interpolation between multiple keyframes during an animation time.